server/chat-plugins/room-events.ts

```diff
@@ -19,30 +34,47 @@ function formatEvent(event: RoomEvent, showAliases?: boolean) {
 	if (!isNaN(timeRemaining)) {
 		explanation = `This event will start in: ${Chat.toDurationString(timeRemaining, {precision: 2})}`;
 	}
+
+	const eventID = toID(event.eventName);
+	const aliases = getAllAliases(room).filter(
+		alias => (room.settings.events?.[alias] as RoomEventAlias).eventID === eventID
+	);
+	const categories = getAllCategories(room).filter(
+		category => (room.settings.events?.[category] as RoomEventCategory).events.includes(eventID)
+	);
+
 	let ret = `<tr title="${explanation}">`;
 	ret += Utils.html`<td>${event.eventName}</td>`;
-	ret += showAliases ? Utils.html`<td>${event.aliases?.join(", ")}</td>` : ``;
+	if (showAliases) ret += Utils.html`<td>${aliases.join(", ")}</td>`;
+	if (showCategories) ret += Utils.html`<td>${categories.join(", ")}</td>`;
 	ret += `<td>${Chat.formatText(event.desc, true)}</td>`;
 	ret += Utils.html`<td><time>${event.date}</time></td></tr>`;
 	return ret;
 }
 
 function getAllAliases(room: Room) {
-	if (!room.settings.events) return [];
 	const aliases: string[] = [];
-	for (const event of Object.values(room.settings.events)) {
-		if (event.aliases) aliases.push(...event.aliases);
+	for (const alias of Object.values(room.settings.events!).filter(a => 'eventID' in a).map(a => a as RoomEventAlias)) {
+		aliases.push(alias.id);
 	}
 	return aliases;
 }
+
+function getAllCategories(room: Room) {
+	const categories: string[] = [];
+	for (const category of Object.values(room.settings.events!)
+		.filter(cat => 'events' in cat)
+		.map(cat => cat as RoomEventCategory)
+	) {
```

Comment:
```suggestion
	if (!room.settings.events) return [];
	const categories: string[] = [];
	for (const category of Object.values(room.settings.events).filter(cat => 'events' in cat)) {
```

If this isn't enough for TS, I'd probably define `events?: undefined` on `RoomEvent` and `RoomEventAlias` and then switch this to `cat => cat.events`.